src/Schema/Directives/UpsertDirective.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Nuwave\Lighthouse\Schema\Directives;
+
+use Illuminate\Support\Collection;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\DatabaseManager;
+use Nuwave\Lighthouse\Schema\Values\FieldValue;
+use Nuwave\Lighthouse\Execution\MutationExecutor;
+use Nuwave\Lighthouse\Support\Contracts\GlobalId;
+use Nuwave\Lighthouse\Support\Contracts\FieldResolver;
+use Nuwave\Lighthouse\Support\Contracts\DefinedDirective;
+
+class UpsertDirective extends BaseDirective implements FieldResolver, DefinedDirective
```

Comment:
There is a lot of duplication between this and `@update`, even some with `@create`. I think we should be able to refactor this so only `name()`, `definition()` and a new function like `makeExecutor(Model $model, array $args): \Closure` should be different.

Reply:
There are a lot to refactor from MutationExecutor too, but I don't know if it is the best idea to do it now or it will be a never end refactor. I'll take a look.

Reply:
Let us just do the directives for now, since nested mutations are likely to change soon - making them compose.